exist-core/src/main/java/org/exist/xquery/value/DoubleValue.java

```diff
@@ -168,78 +167,46 @@ public boolean isPositive() {
     @Override
     public AtomicValue convertTo(final int requiredType) throws XPathException {
         switch (requiredType) {
-            case Type.ATOMIC:
-            case Type.ITEM:
-            case Type.NUMBER:
-            case Type.DOUBLE:
+            case Type.ATOMIC, Type.ITEM, Type.NUMBER, Type.DOUBLE -> {
                 return this;
-            case Type.FLOAT:
-                //if (Float.compare(value, 0.0f) && (value < Float.MIN_VALUE || value > Float.MAX_VALUE)
-                //	throw new XPathException(getExpression(), "Value is out of range for type xs:float");
-                //return new FloatValue((float) value);
+            }
+            case Type.FLOAT -> {
```

Comment:
Same comment as above... and throughout this `switch` statement!

Reply:
Same as above..

Reply:
I like it, too.